Sources/Workspace/WorkspaceConfiguration.swift

```diff
@@ -209,6 +209,118 @@ extension Workspace {
+            case .workspaceAndUser(let rootPath):
+                // package/project "local" .netrc file, takes priority over user-level file
+                let localPath = rootPath.appending(component: ".netrc")
+                // user didn't tell us to explicitly use these .netrc files so be more lenient with errors
+                if let localProvider = self.loadOptionalNetrc(path: localPath, fileSystem: fileSystem, observabilityScope: observabilityScope) {
+                    providers.append(localProvider)
+                }
+
+                // user .netrc file (most typical)
+                let userHomePath = fileSystem.homeDirectory.appending(component: ".netrc")
+                // user didn't tell us to explicitly use these .netrc files so be more lenient with errors
+                if let userHomeProvider = self.loadOptionalNetrc(path: userHomePath, fileSystem: fileSystem, observabilityScope: observabilityScope) {
+                    providers.append(userHomeProvider)
+                }
+            case .user:
+                // user .netrc file (most typical)
+                let userHomePath = fileSystem.homeDirectory.appending(component: ".netrc")
+                
+                // user didn't tell us to explicitly use these .netrc files so be more lenient with errors
+                if let userHomeProvider = self.loadOptionalNetrc(path: userHomePath, fileSystem: fileSystem, observabilityScope: observabilityScope) {
+                    providers.append(userHomeProvider)
+                }
+            case .disabled:
+                // noop
+                break
+            }
+
+            switch self.keychain {
+            case .enabled:
+                #if canImport(Security)
+                providers.append(KeychainAuthorizationProvider(observabilityScope: observabilityScope))
+                #else
+                assertionFailure("Keychain not supported on this platform")
+                #endif
+            case .disabled:
+                // noop
+                break
+            }
+
+            return providers.isEmpty ? .none : CompositeAuthorizationProvider(providers, observabilityScope: observabilityScope)
+        }
+
+        private func loadOptionalNetrc(
+            path: AbsolutePath,
+            fileSystem: FileSystem,
+            observabilityScope: ObservabilityScope
+        ) -> NetrcAuthorizationProvider? {
+            guard fileSystem.exists(path) else {
+                return .none
+            }
+
+            do {
+                // FIXME: need new API in FileSystem to test for read and write access
+                // test for read access
+                try _ = fileSystem.getDirectoryContents(path.parentDirectory)
+            } catch {
+                observabilityScope.emit(warning: "\(path.parentDirectory) is not accessible or not readable, not using .netrc file in it. \(error)")
```

Comment:
Replicating Boris' suggestion:

```suggestion
                observabilityScope.emit(warning: "\(path.parentDirectory) is not accessible or not readable, not using .netrc file in it: \(error)")
```